Add validation case val-2g for deuterium transport in PCC (#357)
idaholab/TMAP8: doc/content/verification_and_validation/val-2g.md

# val-2g

# Deuterium Transport in Proton-Conducting Ceramics

## Case Description

> This validation case involves deuterium gas (D$_2$) and heavy water (D$_2$O) transport in proton-conducting ceramics (PCC), specifically yttrium-doped barium zirconate (BaZr$_{0.9}$Y$_{0.1}$O$_{2.95}$, also known as BZY10). The experimental data used for validation comes from thermal desorption spectroscopy (TDS) measurements performed by Hossain et al. as reported in [!cite](yang2026elucidating) and [!cite](hossain2022comparative). The primary objective is to understand and model the mechanisms of hydrogen isotope transport in PCC materials, which are of interest for tritium extraction systems in fusion energy applications.

Reviewer: ```suggestion
This case reproduces the analysis published in [!cite](yang2026elucidating), which proposed a new model that accurately reproduces deuterium transport in a proton-conducting ceramics in both dry and wet environment. 
This validation effort involves deuterium gas (D$_2$) and heavy water (D$_2$O) transport in proton-conducting ceramics (PCC), specifically yttrium-doped barium zirconate (BaZr$_{0.9}$Y$_{0.1}$O$_{2.95}$, also known as BZY10). The experimental data used for validation comes from thermal desorption spectroscopy (TDS) measurements performed by Hossain et al. [!citep](hossain2022comparative). The primary objective is to understand and model the mechanisms of hydrogen isotope transport in PCC materials, which are of interest for tritium extraction systems in fusion energy applications.
```

The temperature and pressure histories are shown in [val-2g_environment_history].

!media comparison_val-2g.py
    image_name=val-2g_environment_history.png
    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_environment_history
    caption=Temperature and pressure history during the TDS simulation for the dry (D$_2$) and wet (D$_2$O) condition.

## Model Description

### Diffusion of Mobile Species

In BZY10, three mobile species are considered: protonic defects (OD$^{\bullet}$), oxygen vacancies (V$_{\text{O}}^{\bullet\bullet}$), and electrons (e$'$). The hydrogen isotope transport model for PCCs considers diffusion, trapping within the material, and kinetic chemical reactions on the surface. These transport mechanisms are illustrated in [val-2g_transport_schematic].

!media figures/val-2g_transport_schematic.png
    style=width:35%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_transport_schematic
    caption=Schematic of the hydrogen isotope transport in PCCs.

The transport of each species follows Fick's law:

\begin{equation}
\frac{\partial C_i}{\partial t} = \nabla \cdot (D_i \nabla C_i)

> \end{equation}

Reviewer: This is not accurate, the model uses the McNabb and Foster equation. You are somehow not accounting for trapping here.

Reviewer: Also, the concentration of the mobile species is not consistent with the one used in the trapping equation. 
I do not understand why you simply do not use the equations from the paper here.

!media comparison_val-2g.py
    image_name=val-2g_wet_trapping_flux_comparison.png
    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_wet_trapping_flux_comparison
    caption=Comparison of TMAP8 calculations (with trapping, initial parameters) with experimental data for the wet condition. RMSPE values are shown on the plot.

### Results With Trapping Using Calibrated Parameters

To improve the agreement with experimental data, the model parameters were calibrated using the [Parallel Subset Simulation](samplers/ParallelSubsetSimulation.md) (PSS) approach in [MOOSE's stochastic tools module](modules/stochastic_tools/index.md). The PSS method systematically explores the parameter space to minimize the difference between simulation predictions and experimental measurements. A detailed explanation of this method is provided in [!cite](yang2026elucidating). In this validation case, a simplified PSS optimization process is used with initial parameters close to the calibrated values to reduce computational cost. The evolution of the objective function (log inverse error) is shown below:

> !media comparison_val-2g.py

Reviewer: This figure is not extremely interesting as is.

    image_name=val-2g_trapping_optimization_PSS_iterations.png
    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_trapping_optimization_PSS_iterations
    caption=The evolution of the objective function (log inverse error) during PSS optimization.

> [val-2g_trapping_optimization_inputs_PSS] shows the deviation of all calibrated parameters from the initial parameters. All deviations are within two standard deviations of their corresponding normal distribution ranges.

Reviewer: `All deviations are within two standard deviations of their corresponding normal distribution ranges.` is not true here.

    caption=Comparison of TMAP8 calculations (with trapping, calibrated parameters) with experimental data for the wet condition. RMSPE values are shown on the plot.

The calibrated model successfully reproduces the key features of the TDS spectra:

- The peak temperature positions for both D$_2$ and D$_2$O desorption
- The relative magnitudes of the desorption peaks
- The peak shapes and widths

This validation demonstrates TMAP8's capability to model hydrogen isotope transport in proton-conducting ceramics, including the complex interplay between diffusion, trapping, and surface reactions.

## Input files

!style halign=left
The input files for this validation case are:

- [/val-2g_trapping_initial_parameters.i]: Simulates deuterium transport with and without trapping effects using corresponding parameters. [/parameters_no_trapping_initial_validation.params] includes the initial parameters for simulation without trapping, [/parameters_trapping_initial_validation.params] includes the initial parameters for simulation with trapping, and [/parameters_trapping_calibrated_validation.params] includes the calibrated parameters for simulation with trapping.

> - [/val-2g_trapping_light.i]: Simulates a version of the file with a coarser mesh, fewer time steps, no trapping effects, and wet environment to limit the computational costs.

Reviewer: ```suggestion
- [/val-2g_trapping_light.i]: Simulates a version of the file with a coarser mesh, fewer time steps, no trapping effects, and wet environment to limit the computational costs for testing purposes.
```

The temperature and pressure histories are shown in [val-2g_environment_history].

!media comparison_val-2g.py
    image_name=val-2g_environment_history.png
    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_environment_history
    caption=Temperature and pressure history during the TDS simulation for the dry (D$_2$) and wet (D$_2$O) condition.

## Model Description

### Diffusion of Mobile Species

In BZY10, three mobile species are considered: protonic defects (OD$^{\bullet}$), oxygen vacancies (V$_{\text{O}}^{\bullet\bullet}$), and electrons (e$'$). The hydrogen isotope transport model for PCCs considers diffusion, trapping within the material, and kinetic chemical reactions on the surface. These transport mechanisms are illustrated in [val-2g_transport_schematic].

!media figures/val-2g_transport_schematic.png
    style=width:35%;margin-bottom:2%;margin-left:auto;margin-right:auto
    id=val-2g_transport_schematic
    caption=Schematic of the hydrogen isotope transport in PCCs.

The governing equations for OD$^{\bullet}$ in PCC materials is described as

\begin{equation}

> \frac{\partial C_M}{\partial t} = \nabla D \nabla C_M - \frac{\partial C_T}{\partial t},

Reviewer: ```suggestion
\frac{\partial C_M}{\partial t} = \nabla \cdot D \nabla C_M - \frac{\partial C_T}{\partial t},
```
